src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -50,18 +50,23 @@ public static BooleanExpression isAnyFieldSetForSelectedEntry(List<Field> fields
     }
 
     public static BooleanExpression isFilePresentForSelectedEntry(StateManager stateManager, PreferencesService preferencesService) {
-        return Bindings.createBooleanBinding(() -> {
-                    List<LinkedFile> files = stateManager.getSelectedEntries().get(0).getFiles();
-                    if ((files.size() > 0) && stateManager.getActiveDatabase().isPresent()) {
-                        Optional<Path> filename = FileHelper.find(
-                                stateManager.getActiveDatabase().get(),
-                                files.get(0).getLink(),
-                                preferencesService.getFilePreferences());
-                        return filename.isPresent();
-                    } else {
-                        return false;
-                    }
-                }, stateManager.getSelectedEntries(),
-                stateManager.getSelectedEntries().get(0).getFieldBinding(StandardField.FILE));
+
+        ObservableList<BibEntry> selectedEntries = stateManager.getSelectedEntries();
+        MonadicObservableValue<Boolean> fileIsPresent = EasyBind.monadic(Bindings.valueAt(selectedEntries, 0)).map(entry -> {
+            List<LinkedFile> files = entry.getFiles();
```

Comment:
I think this does not update the status if a file is added to the currently selected entry. For this, the isAnyFieldSetForSelectedEntry method uses `flatMap` instead of `map`.

Reply:
I just tested it, it works as well when I add or remove a file to an entry both the context menu and the View menu get updated with the correct status